src/class/Cars.js

```diff
@@ -1,60 +1,133 @@
 import Car from "./Car";
 
-const CAR_ADVANCE_MAX_NUMBER = 9;
-
-const CAR_ADVANCE_THRESHOLD_NUMBER = 4;
+export default class Cars {
+  static #DEFAULT_RACING_ROUND_NUMBER = 5;
+  static #RACING_ROUND_MIN_VALUE = 1;
+  static GAME_STEP = Object.freeze({
+    SET_CARS: "SET_CARS",
+    SET_ROUND_NUMBER: "SET_ROUND_NUMBER",
+    EXECUTE_ROUND: "EXECUTE_ROUND",
+    AWARDS: "AWARDS",
+  });
+  static ERROR_MESSAGES = Object.freeze({
+    DUPLICATE_CAR_NAME: "자동차 이름은 중복될 수 없습니다.",
+    INVALID_RACING_ROUND_FORMAT: "양의 정수 형식의 값을 입력해 주세요.",
+    INVALID_RACING_ROUND_VALUE: "1이상 값을 입력해주세요.",
+  });
+  static #rRacingRound = /^[0-9]+$/;
+  static #CAR_ADVANCE_MAX_NUMBER = 9;
+  static #CAR_ADVANCE_THRESHOLD_NUMBER = 4;
+  static #defaultAdvanceCondition = () => {
+    return (
+      Math.random() * this.#CAR_ADVANCE_MAX_NUMBER >=
+      this.#CAR_ADVANCE_THRESHOLD_NUMBER
+    );
+  };
 
-const DEFAULT_RACING_ROUND_NUMBER = 5;
+  #roundNumber = Cars.#DEFAULT_RACING_ROUND_NUMBER;
+  #cars = [];
+  #nextGameStep = Cars.GAME_STEP.SET_CARS;
 
-const defaultAdvanceCondition = () => {
-  return Math.random() * CAR_ADVANCE_MAX_NUMBER >= CAR_ADVANCE_THRESHOLD_NUMBER;
-};
+  constructor(names) {
+    if (
+      names instanceof Array &&
```

Comment:
```suggestion
      Array.isArary(names) &&
```
isArray가 커버하는 영역이 더 많은걸로 알고 있어요!

Reply:
오 그렇군요~

잘 몰랐던 부분인데 자세히 찾아봐야겠네요ㅎㅎ